neo4j/internal/bolt/bolt4.go

```diff
@@ -618,13 +618,19 @@ func (b *bolt4) GetRoutingTable(database string, context map[string]string) (*db
 		return nil, err
 	}
 
+	// The query should run in system database, preserve current setting and restore it when
+	// done.
+	originalDatabaseName := b.databaseName
+	b.databaseName = "system"
```

Comment:
Shouldn't this be the passed in variable `database`?

The routing table will be different for each database, so I don't think just `system` is the right call here. I guess we need to something like:

`b.databaseName = database` with the added deferred func to restore the previous one.

Reply:
b.databaseName is the database that is used to execute within, there is a parameter to getRoutingTable that is set to database, that is we query for routing for "database" in "system"
It would be great if you could verify it with your setup. I'm working on another patch to include in the release.